connman/src/dns-systemd-resolved.c

```diff
@@ -375,14 +376,29 @@ int __connman_dnsproxy_append(int index, const char *domain,
 
 int __connman_dnsproxy_add_listener(int index)
 {
-	DBG("");
+	int lo_index;
+	int err;
+
+	DBG("index %d", index);
 
-	return -ENXIO;
+	err = dns_add_listener(index, DNS_IPPROTO_UDP, true);
+	if (err)
+		return err;
+
+	lo_index = connman_inet_ifindex("lo");
+	err = dns_create_server(lo_index, "local", "127.0.0.53", IPPROTO_UDP,
+									true);
```

Comment:
Maybe there could be a `struct server_data*` handle returned (and need to rename those as they're exposed) and a call `dns_server_set_enabled(bool enable)`...